// Copyright 2019 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe.autoflip;

import "mediapipe/framework/calculator.proto";

message LipTrackCalculatorOptions {
  extend mediapipe.CalculatorOptions {
    optional LipTrackCalculatorOptions ext = 284226724;
  }

  // Number of previous frames stored which are used to calculate
  // the lip statistics. Note that variance_history should be no
  // smaller than mean_history.
  optional int32 mean_history = 1 [default = 2];
  optional int32 variance_history = 2 [default = 6];

  // If IOU of two face bounding boxes are greater than iou_threshold,
  // these two faces might be considered as the same face. Otherwise,
  // these two faces are not considered as the same face.
  optional float iou_threshold = 3 [default = 0.5];

  // Lip statistics thresholds which are used to detect
  // the active speaker.
  optional float lip_inner_mean_threshold_big_mouth = 4 [default = 0.18];
  optional float lip_inner_variance_threshold_big_mouth = 5 [default = 0.001];
  optional float lip_inner_mean_threshold_small_mouth = 6 [default = 0.07];
  optional float lip_inner_variance_threshold_small_mouth = 7 [default = 0.00125];
  optional float lip_outer_mean_threshold_big_mouth = 8 [default = 0.7];
  optional float lip_outer_variance_threshold_big_mouth = 9 [default = 0.0015];
  optional float lip_outer_mean_threshold_small_mouth = 10 [default = 0.6];
  optional float lip_outer_variance_threshold_small_mouth = 11 [default = 0.0020];

  // Output shot boundarys
  optional bool output_shot_boundary = 12 [default = true];
  // Only send shot boundary results if the speaker changes
  optional bool output_shot_boundary_only_on_change = 13 [default = true];
  // Minimum number of shot duration (in seconds).
  optional double min_shot_span = 14 [default = 1];

  // Minimum number of speaker duration (in microseconds).
  optional double min_speaker_span = 15 [default = 2500000];
}
